daemon/checkpoint.go

```diff
@@ -16,6 +16,41 @@ var (
 	validCheckpointNamePattern = api.RestrictedNamePattern
 )
 
+// getCheckpointDir verifies checkpoint directory for create,remove, list options and checks if checkpoint already exists
+func getCheckpointDir(checkDir, checkpointID string, name string, containerID string, containercheckpointDir string, create bool) (checkpointDir string, err error) {
```

Comment:
Looks like the output variables aren't really used; this could be

```go
func getCheckpointDir(checkDir, checkpointID string, name string, containerID string, containercheckpointDir string, create bool) (string, error) { 
	var checkpointDir string
	var errorVar error
```

In addition, I would;

- change `errorVar` to a different name (`Var` doesn't add much value); perhaps simply `err2`
- change the `name` argument to something that explains it's the _container_'s name, not the checkpoint
- `containercheckpointDir` should use camelcase, `containerCheckpointDir`

(perhaps use `ctrName`, `ctrID`, `ctrCheckpointDir`)


I've looked at possibly use not that many arguments, and instead something like;

```go
func getCheckpointDir(ctr *container.Container, checkpoint types.CheckpointOptions, create bool) (string, error) {
```

Which would make it a lot cleaner, but that's a bigger change, so not something for this PR